### PR TITLE
Minor spec fix : correct variadic result notation for all_gather and all_reduce ops

### PR DESCRIPTION
**Pull request by abhigunj**

### CODE REVIEW

**Review by ghpvnist — APPROVED** [review]

File: docs/spec.md [hunk 1]
```diff
@@ -823,7 +823,7 @@ Afterwards, within each `process_group`:
 // %operand0@(1, 0): [5, 6, 7, 8]
 // %operand1@(0, 0): [9, 10, 11, 12]
 // %operand1@(1, 0): [13, 14, 15, 16]
-%result = "stablehlo.all_reduce"(%operand0, %operand0) ({
+%result:2 = "stablehlo.all_reduce"(%operand0, %operand0) ({
```
Comment (sdasgup3) [inline docs/spec.md:826]: Also change the examples in StableHLOOps.td

**Review by sdasgup3 — APPROVED** [review]

File: docs/spec.md (hunk 1 above)
Comment (abhigunj) [reply docs/spec.md:826]: StableHLOOps.td examples are non variadic, don't need the change.

Comment (sdasgup3) [reply docs/spec.md:826]: [Something to consider for a follow up]
Can we have the same example in both places as done for other ops?

Comment (ghpvnist) [reply docs/spec.md:826]: Let's keep the example same per 7.v in https://github.com/openxla/stablehlo/blob/main/docs/reference_checklist.md#after-implementing-the-op.

Comment (abhigunj) [reply docs/spec.md:826]: https://github.com/openxla/stablehlo/pull/2481  will match the examples

**Pull request merged**